File: 101/exercises/easy7/lettercasecounter.rb

def letter_case_count(str)
  counts = { lowercase: 0, uppercase: 0, neither: 0 }
  # str.each_char do |char|
  #   if char.swapcase == char
  #     counts[:neither] += 1
  #   elsif char.upcase == char
  #     counts[:uppercase] += 1
  #   elsif char.downcase == char
  #     counts[:lowercase] += 1
  #   end
  # end
  counts[:uppercase] = str.scan(/[A-Z]/).size
  counts[:lowercase] = str.scan(/[a-z]/).size
  counts[:neither] = str.length - counts[:uppercase] - counts[:lowercase]
  counts
end

## Tests

p letter_case_count('abCdef 123') == { lowercase: 5, uppercase: 1, neither: 4 }
p letter_case_count('AbCd +Ef') == { lowercase: 3, uppercase: 3, neither: 2 }
p letter_case_count('123') == { lowercase: 0, uppercase: 0, neither: 3 }
p letter_case_count('') == { lowercase: 0, uppercase: 0, neither: 0 }

File: 210/small-problems/text-processing/lettercase-counter.js
function letterCaseCount(string) {
  var lowercaseCount = (string.match(/[a-z]/g) || []).length;
  var uppercaseCount = (string.match(/[A-Z]/g) || []).length;

  return {
           lowercase: lowercaseCount,
           uppercase: uppercaseCount,
           neither: string.length - uppercaseCount - lowercaseCount
         };
}

console.log(letterCaseCount('abCdef 123'));  // { lowercase: 5, uppercase: 1, neither: 4 }
console.log(letterCaseCount('AbCd +Ef'));    // { lowercase: 3, uppercase: 3, neither: 2 }
console.log(letterCaseCount('123'));         // { lowercase: 0, uppercase: 0, neither: 3 }
console.log(letterCaseCount(''));            // { lowercase: 0, uppercase: 0, neither: 0 }
